### DISCUSSION TITLE
Is it possible to use timewindows as soft constraints?

### DISCUSSION THREAD

**Discussion by MxNxP** [reinterpretcat/vrp#93] [Q&A] [answered]
We have delivery jobs with timewindows and we need to complete all the jobs even when some of the timewindows are violated. Is it possible to calculate solutions that do not strictly take the time windows into account, but allow early / late arrivals if necessary for jobs that would otherwise be unassigned?

**Comment by reinterpretcat** [comment, marked as answer]
No, at this point time windows are hard constraints and their violations is not allowed. However, I think it is possible to implement such requirement as a new soft constraint. This would require:
- change format to support soft time window definition
- add a new soft constraint which penalizes late or early arrivals outside of strict time windows
- add a new custom objective which prefers solutions with less soft constraint violations

**Reply by MxNxP** [reply]
Thank you for your quick response!
